### PR TITLE
Replaced Codecov with Coveralls.

### PR DESCRIPTION
**Pull request by binary1248**
Title.

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -179,12 +179,28 @@ if(SFML_ENABLE_COVERAGE AND OpenCppCoverage_FOUND)
...
+    string(REPLACE "/" "\\\\" COVERAGE_PATH_PREFIX "${PROJECT_SOURCE_DIR}/")
+    string(REPLACE "${COVERAGE_ROOT_NAME}\\\\" "" COVERAGE_PATH_PREFIX "${COVERAGE_PATH_PREFIX}")
+
+    file(WRITE "${PROJECT_BINARY_DIR}/patch_coverage.cmake"
+         "file(READ \"${PROJECT_BINARY_DIR}/coverage.out\" COVERAGE_OUT)\n\
+         string(REPLACE \"${COVERAGE_PATH_PREFIX}\" \"\" COVERAGE_OUT \"\${COVERAGE_OUT}\")\n\
+         string(REPLACE \"${COVERAGE_ROOT_NAME}\" \".\" COVERAGE_OUT \"\${COVERAGE_OUT}\")\n\
+         file(WRITE \"${PROJECT_BINARY_DIR}/coverage.out\" \"\${COVERAGE_OUT}\")\n")
+
+    set(COVERAGE_PREFIX ${OpenCppCoverage_BINARY} --quiet --export_type cobertura:${PROJECT_BINARY_DIR}/coverage.out --cover_children --excluded_modules "${COVERAGE_EXCLUDE}" --sources "${COVERAGE_SRC}" --sources "${COVERAGE_INCLUDE}" --)
```
Comment (eXpl0it3r) [inline test/CMakeLists.txt:193]: According to the command line reference, the `--export_type` and `--sources` should use `=`: https://github.com/OpenCppCoverage/OpenCppCoverage/wiki/Command-line-reference

When I [tried it with Codecov](https://github.com/SFML/SFML/tree/bugfix/opencppcoverage), I got different results, but not sure, but Codecov couldn't understand them.

Comment (binary1248) [reply test/CMakeLists.txt:193]: If you look at the source, they don't make a difference between these arguments and the others.
Also, when running it with `--verbose` it seems to recognize the arguments... so... 🤷

**Review by eXpl0it3r — APPROVED** [review]
Looks fine to me. Since @ChrisThrasher has been mostly working with Codecov, I'd be interested in hearing his opinion on switchting to coveralls.

We may also want to enable PR comments like before, even if it annoyed some people in the past, it's at least much easier to see the complete result and/or jump into the coverage report page.

https://docs.coveralls.io/app-notifications

**Comment by binary1248** [comment]
PR comments are enabled, but according to what I could find on their issue trackers, their bot only creates a PR comment if it has coverage for a base branch it can compare against. Since this is the first revision for which coverage is uploaded to Coveralls the bot won't create a comment in this PR. Weird implementation, but that's how it is. 🤷

**Review by ChrisThrasher — APPROVED** [review]
I clicked through the coverage report and it seems correct as far as I can tell. I'm not familiar with Coveralls but I'm fine with us trying it out.

**Pull request merged**